ui/segment.reel/segment.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @module ui/segmented-bar.reel/segment.reel
+ * @requires montage/ui/component
+ */
+var Component = require("ui/component").Component;
+
+/**
+ * @class Segment
+ * @classdesc
+ * A segment to be used inside a segmented bar.
+ * Uses FlexBox to determine rendered length.
+ *
+ * @extends Component
+ * @see SegmentedBar
+ */
+exports.Segment = Component.specialize( /** @lends Segment.prototype # */ {
+    /**
+     * @private
+     * @prop {Object} value
+     * @prop {Number} value.length
+     * @prop {String} value.label
+     */
+    _content: {value: null},
+
+    /**
+     * Length is used in draw()
+     * Label is bound to Text component in template
+     *
+     * E.g. usage to get content from SegmentedBar via binding,
+     * where `region` and `size` keys can be named anything in your data:
+     *
+     * ```json
+     * "content": {
+     *     "<-": "@segmentedBar:segment.object.{label: region, length: size}"
+     * }
+     * ```
+     *
+     * @param {Object} newContent
+     * @prop {Number} newContent.length
+     * @prop {String} newContent.label
+     * @returns {Object[]}
+     */
+    content: {
+        get: function () {return this._content},
+        set: function (newContent) {
+            if (newContent !== this._content) {
+                this._content = newContent;
+                this.needsDraw = true;
+            }
+        }
+    },
+
+    /**
+     * Redraw based on updated length
+     *
+     * @method
+     */
+    draw: {value: function () {
+        if (this.content) {
+            this.element.style.webkitBoxFlex = this.content.length;
+            this.element.style.webkitFlex = this.content.length;
+            this.element.style.msFlex = this.content.length;
+            this.element.style.flex = this.content.length;
+        }
+    }}
+});
```

Comment:
This is what we agreed on:

> content to data; contentController to dataController; iteration.object to iteration.data

Should we start changing now or wait for Repetition to be changed first? If we start now we still have to use iteration.object. I'm ok either way. Your call.